src/components/cluster-deployments/ClusterDeploymentDetailsPage.tsx

```diff
@@ -0,0 +1,172 @@
+};
+
+const getClusterDeploymentActions =
+  (agentClusterInstall?: CIM.AgentClusterInstallK8sResource): KebabOptionsCreator =>
+  (kindObj: K8sKind, clusterDeployment: K8sResourceCommon) => {
+    const { namespace, name } = clusterDeployment.metadata || {};
+    return [
+      {
+        label: 'Edit',
+        href: `/k8s/ns/${namespace}/${ClusterDeploymentKind}/${name}/edit`,
+        isDisabled: !canEditCluster(agentClusterInstall),
+      },
+    ];
+  };
+
+export const ClusterDeploymentOverview = (props: DetailsTabProps) => {
+  const { obj: clusterDeployment } = props;
+  const [agentClusterInstall, agentClusterInstallLoaded, agentClusterInstallError] =
+    useK8sWatchResource<CIM.AgentClusterInstallK8sResource>(
+      clusterDeployment?.spec?.clusterInstallRef.name
+        ? {
+            kind: AgentClusterInstallKind,
+            name: clusterDeployment?.spec?.clusterInstallRef.name,
+            namespace: clusterDeployment?.metadata?.namespace,
+            namespaced: true,
+            isList: false,
+          }
+        : undefined,
+    );
+
+  const agentSelector = clusterDeployment.spec?.platform?.agentBareMetal?.agentSelector;
+  const [agents, agentsLoaded, agentsError] = useK8sWatchResource<CIM.AgentK8sResource[]>(
+    agentSelector
+      ? {
+          kind: AgentKind,
+          isList: true,
+          selector: agentSelector,
+          namespaced: true,
+        }
+      : undefined,
+  );
+
+  if (!clusterDeployment) return null;
+  if (agentsError) throw new Error(agentsError);
+  if (agentClusterInstallError) throw new Error(agentClusterInstallError);
+  if (!(agentsLoaded && agentClusterInstallLoaded)) return <LoadingState />;
+
+  const handleKubeconfigDownload = async () => {
+    const kubeconfigSecretName =
+      agentClusterInstall.spec?.clusterMetadata?.adminKubeconfigSecretRef.name;
+    const kubeconfigSecretNamespace = clusterDeployment.metadata.namespace;
+    agentClusterInstall.spec?.clusterMetadata?.adminKubeconfigSecretRef.name;
+    try {
+      const kubeconfigSecret = await k8sGet(
+        SecretModel,
+        kubeconfigSecretName,
+        kubeconfigSecretNamespace,
+      );
+      const blob = new Blob([atob(kubeconfigSecret.data.kubeconfig)], {
+        type: 'text/plain;charset=utf-8',
+      });
+      saveAs(blob, 'kubeconfig.json');
+    } catch (e) {
+      console.error('Failed to fetch kubeconfig secret.', e);
+    }
+  };
+
+  const handleFetchEvents = async (props, onSuccess, onError) => {
+    try {
+      const res = await fetch(agentClusterInstall.status?.debugInfo?.eventsURL, {
+        mode: 'same-origin',
+      });
+      const data = await res.json();
+      onSuccess(
+        data.map((event: any) => ({
+          clusterId: event.cluster_id,
+          eventTime: event.event_time,
+          message: event.message,
+          severity: event.severity,
+        })),
+      );
+    } catch (e) {
+      onError('Failed to fetch cluster events.');
+    }
+  };
```

Comment:
I think this can also be replaced by something like `type fetchEvents = (url: string) => Promise<any>`

Reply:
I've moved whole events fetching to UI lib an will eventually add fetcher function as prop in a follow-up.